internals/crypto/rsa.go

```diff
@@ -358,73 +358,88 @@ type CiphertextRSAAES struct {
 	rsa CiphertextRSA
 }
 
-// MarshalJSON encodes the ciphertext in a string.
-func (ct CiphertextRSAAES) MarshalJSON() ([]byte, error) {
+// EncodeToString encodes the ciphertext in a string.
+func (ct CiphertextRSAAES) EncodeToString() string {
 	data := base64.StdEncoding.EncodeToString(ct.aes.Data)
 
 	metadata := newEncodedCiphertextMetadata(map[string]string{
 		"nonce": base64.StdEncoding.EncodeToString(ct.aes.Nonce),
 		"key":   base64.StdEncoding.EncodeToString(ct.rsa.Data),
 	})
 
-	return json.Marshal(fmt.Sprintf("%s$%s$%s", algorithmRSAAES, data, metadata))
+	return fmt.Sprintf("%s$%s$%s", algorithmRSAAES, data, metadata)
```

Comment:
Tiny note: using `Sprintf` for simple string concatenation is pretty 'slow' when compared to `algorithmRSAAES+"$"+data+"$"+metadata` or using something like `StringBuilder`.

Totally not a priority right now, but it's worth checking out:

https://gist.github.com/dtjm/c6ebc86abe7515c988ec

https://stackoverflow.com/questions/1760757/how-to-efficiently-concatenate-strings-in-go/47798475#47798475

Reply:
:+1: 

Let's pick this up in a new PR, as this PR does not change the status-quo.